google/cloud/bigtable/backup.py

```diff
@@ -0,0 +1,332 @@
+        :raises: ValueError: If the backup name does not match the expected
+                             format or the parsed project ID does not match the
+                             project ID on the Instance's client, or if the
+                             parsed instance ID does not match the Instance ID.
+        """
+        match = re.compile(
+            r"^projects/(?P<project>[^/]+)/"
+            r"instances/(?P<instance_id>[a-z][-a-z0-9]*)/"
+            r"clusters/(?P<cluster_id>[a-z][-a-z0-9]*)/"
+            r"backups/(?P<backup_id>[a-z][a-z0-9_\-]*[a-z0-9])$"
+        ).match(backup_pb.name)
+        if match is None:
+            raise ValueError(
+                "Backup protobuf name was not in the expected format.", backup_pb.name
+            )
+        if match.group("project") != instance._client.project:
+            raise ValueError(
+                "Project ID of the Backup does not match the Project ID "
+                "of the instance's client"
+            )
+
+        instance_id = match.group("instance_id")
+        if instance_id != instance.instance_id:
+            raise ValueError(
+                "Instance ID of the Backup does not match the Instance ID "
+                "of the instance"
+            )
+        backup_id = match.group("backup_id")
+        return cls(backup_id, instance)
+
+    def create(self, cluster_id=None):
+        """Creates this backup within its instance.
+
+        :type cluster_id: str
+        :param cluster_id: (Optional) The ID of the Cluster for the newly
+                           created Backup.
+
+        :rtype: :class:`~google.api_core.operation.Operation`
+        :returns: :class:`~google.cloud.bigtable_admin_v2.types._OperationFuture`
+                  instance, to be used to poll the status of the 'create' request
+        :raises Conflict: if the Backup already exists
+        :raises NotFound: if the Instance owning the Backup does not exist
+        :raises BadRequest: if the `table` or `expire_time` values are invalid,
+                            or `expire_time` is not set
+        """
+        if not self._expire_time:
+            raise ValueError('"expire_time" parameter must be set')
+            # TODO: Consider implementing a method that sets a default value of
+            #  `expire_time`, e.g. 1 week from the creation of the Backup.
+        if not self.table_id:
+            raise ValueError('"table" parameter must be set')
+
+        if cluster_id:
+            self._cluster = cluster_id
+
+        if not self._cluster:
+            raise ValueError('"cluster" parameter must be set')
+
+        backup = {
+            "source_table": self.source_table,
+            "expire_time": _datetime_to_pb_timestamp(self.expire_time),
+        }
+
+        api = self._instance._client.table_admin_client
+        return api.create_backup(self.parent, self.backup_id, backup)
+
+    def get(self):
```

Comment:
Matter of opinion,
Solely for the sake of consistency, perhaps this method should be `reload`, similar to [`instance.reload`](https://github.com/googleapis/python-bigtable/blob/5c1d61827618d254c453b3871c0022a8d35bfbb2/google/cloud/bigtable/instance.py#L346), [`cluster.reload`](https://github.com/googleapis/python-bigtable/blob/5c1d61827618d254c453b3871c0022a8d35bfbb2/google/cloud/bigtable/cluster.py#L201)

where `reload` method fetches the object metadata from the server => updates self. Propagates any error to user.

If deciding to update `self`, could re-use `update_from_pb` method suggested above.